src/main/scala/jio/package.scala

```diff
@@ -74,7 +64,37 @@ package object jio extends JioFiles with DecorateAsScala with DecorateAsJava {
     }
   }
 
-  trait Pathish[Rep] {
+  trait Metadataish {
+    def exists: Boolean
```

Comment:
There's no point in checking if it exists before deleting it, because it can cease to exist between the check and the delete. A snapshot of the metadata from some time in the past definitely can't tell you whether it exists. It only can tell you whether it once existed.

That's what the metadata class represents, a metadata snapshot taken at some moment in time. "NoMetadata" is what you have if that snapshot can't be taken for whatever reason. Whether metadata exists at THIS moment in time is another question entirely.